cli/actions/dev.js

```diff
@@ -63,26 +63,32 @@ module.exports = (config) => {
   const compileServer = () => serverCompiler.run(() => undefined);
 
   // Watch the server files and recompile and restart on changes.
-  const watcher = chokidar.watch([serverSrcPath]);
-  watcher.on('ready', () => {
-    watcher
-      .on('add', compileServer)
-      .on('addDir', compileServer)
-      .on('change', compileServer)
-      .on('unlink', compileServer)
-      .on('unlinkDir', compileServer);
-  });
+  if (!noServer) {
+    const watcher = chokidar.watch([serverSrcPath]);
+    watcher.on('ready', () => {
+      watcher
+        .on('add', compileServer)
+        .on('addDir', compileServer)
+        .on('change', compileServer)
+        .on('unlink', compileServer)
+        .on('unlinkDir', compileServer);
+    });
+  }
 
   // Compile Client Webpack Config
   clientCompiler = webpackCompiler(clientConfig, () => {
     afterClientCompile();
-    compileServer();
+    if (!noServer) {
+      compileServer();
+    }
   });
 
   // Compile Server Webpack Config
-  serverCompiler = webpackCompiler(serverConfig, once(() => {
-    ifPortIsFreeDo(serverURL.port, startServer);
-  }));
+  if (!noServer) {
```

Comment:
Maybe we should add the watcher code above into this block so we can group the features under one `if` statement.